### PR TITLE
Add support for importing entities.

### PR DESCRIPTION
**Pull request by keyurva**

### CODE REVIEW

File: simple/tests/stats/test_data/entities_importer/input/without_id_column.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CASE,IUCR,PRIMARY DESCRIPTION,SECONDARY DESCRIPTION
```
Comment (shifucun) [inline simple/tests/stats/test_data/entities_importer/input/without_id_column.csv:1]: why without_id_column.csv the same as with_id_column.csv?

File: simple/stats/config.md [hunk 2]
```diff
@@ -92,6 +120,32 @@ and this field associates one of the provenances defined there to this file.
...
+
+#### `idColumn`
+
+Applicable to events and entities.
+
+Every event and entity is identified by a unique ID.
+If one of the columns encodes that ID, it can be indicated with the `idColumn` field.
+If none is specified, IDs will be auto generated.
+
+#### `entityColumns`
```
Comment (shifucun) [inline simple/stats/config.md:137]: should this call "propertyColumns"?

File: simple/stats/config.md [hunk 3]
```diff
@@ -73,12 +93,20 @@ Example:
...
 
+#### `importType`
+
+This importer supports importing observations, events and entities.
+The type of import can be specified with the `importType` parameter.
+Possible values: `observations` (default), `events`, `entities`.
+
 #### `entityType`
 
 All entities in a given file must be of a specific type. This type should be
```
Comment (shifucun) [inline simple/stats/config.md:104]: nit: "one specific type"

File: simple/stats/config.md [hunk 4]
```diff
@@ -73,12 +93,20 @@ Example:
 
 ### Input file parameters
 
+#### `importType`
+
+This importer supports importing observations, events and entities.
+The type of import can be specified with the `importType` parameter.
+Possible values: `observations` (default), `events`, `entities`.
```
Comment (shifucun) [inline simple/stats/config.md:100]: should the importType be "Entity", "Observation" and "Event"?

File: simple/stats/config.md (hunk 2 above)
Comment (keyurva) [reply simple/stats/config.md:137]: Hmm - all columns are properties so `propertyColumns` won't distinguish between values and ids. Open to other names though - `entityColumns` is not the best name.

File: simple/stats/config.md [hunk 5]
```diff
@@ -92,6 +120,32 @@ and this field associates one of the provenances defined there to this file.
...
+
+Applicable to events and entities.
+
+Every event and entity is identified by a unique ID.
+If one of the columns encodes that ID, it can be indicated with the `idColumn` field.
+If none is specified, IDs will be auto generated.
+
+#### `entityColumns`
+
+Applicable to entities.
```
Comment (shifucun) [inline simple/stats/config.md:139]: what if this is not specified?

File: simple/stats/config.md [hunk 6]
```diff
@@ -132,6 +186,28 @@ Use "/" as a separator to specify a multi-level hierarchy.
 An array of NL sentences to be used for creating more NL embeddings (in addition to the description)
 for the variable.
 
+## `entities`
```
Comment (shifucun) [inline simple/stats/config.md:189]: I lean towards not having this field. If needed, user should just add name and description columns in the csv?

File: simple/stats/config.md (hunk 4 above)
Comment (keyurva) [reply simple/stats/config.md:100]: Don't want to break backward compatibility. So going with what we have.

File: simple/stats/config.md (hunk 3 above)
Comment (keyurva) [reply simple/stats/config.md]: Done.

File: simple/stats/config.md (hunk 5 above)
Comment (keyurva) [reply simple/stats/config.md:139]: Then all columns will be considered object values.

File: simple/stats/config.md (hunk 6 above)
Comment (keyurva) [reply simple/stats/config.md:189]: Can you elaborate - not sure how they can be in the CSV?

Also, this is optional. If they don't provide one, the name will be same as the type.

File: simple/stats/config.md (hunk 4 above)
Comment (shifucun) [reply simple/stats/config.md:100]: As discussed, we should handle both and document the new/desired one.

**Review by shifucun — APPROVED** [review]
LG to unblock. As discussed, we should think through the entire config and come up with a simple solution.

**Comment by keyurva** [comment]
Thanks for the review.

Removed `config.md` from this PR. Bo and I will take a holistic view of the config, make any necessary changes and then checkin the updated doc.

**Pull request merged**